### DISCUSSION TITLE
How to reconnect my client socketIO to my web server using ExpressJS and SocketIO

### DISCUSSION THREAD

**Discussion by Jonathan64340** [socketio/socket.io#4102] [Q&A]
Hello everyone, today i'm needed help from you.
I have making one chat app using socketio front/back.
When i lost connection due to exemple switching from my wifi to my data mobile phone, i am disconnected but when i see in my network i can see the client socketIO try to reconnect on ws://{my_server} but my server using expressJS and i must call only https://{my_server} and not ws://{my_server}

Thank you all for your help !!!!

**Comment by darrachequesne** [comment]
Hi! This is the expected behavior, the Socket.IO client will try to reconnect with either:

- HTTP long-polling, with URL like `https://{my_server}`
- or WebSocket, with URL like `ws://{my_server}`

Please see here: https://socket.io/docs/v4/how-it-works/

Or are you talking about the `ws://` (no TLS) vs `https://` (TLS)?

**Comment by Jonathan64340** [comment]
Hey thank you for your quick response ! My problem is when user is
disconnected i want to reconnect it  to my express server with
https://myserver and when socketio trying to reconnecting it will fetch
ws://myserver but i don't have ws server exposed juste https ....
